src/Microsoft.IdentityModel.Tokens/X509SecurityKey.cs

```diff
@@ -127,6 +127,26 @@ public override SignatureProvider GetSignatureProvider(string algorithm, bool ve
                 return CryptoProviderFactory.CreateForSigning(this, algorithm);
         }
 
+        public override bool IsSupportedAlgorithm(string algorithm)
+        {
+            if (string.IsNullOrEmpty(algorithm))
+                return false;
+
+            switch (algorithm)
+            {
+                case SecurityAlgorithms.RsaSha256:
+                case SecurityAlgorithms.RsaSha384:
+                case SecurityAlgorithms.RsaSha512:
+                case SecurityAlgorithms.RsaSha256Signature:
+                case SecurityAlgorithms.RsaSha384Signature:
+                case SecurityAlgorithms.RsaSha512Signature:
+                    return _certificate.PublicKey != null;
+
+                default:
```

Comment:
Ah fine then. I had not realized ECDsa certificates were not supported by Wilson ATM.

It doesn't change anything to the fact it will return `true` on .NET Desktop if you try to use an ECDsa certificate and specify an RSA algorithm, tho.